internal/clients/nightfall/nightfall_test.go

```diff
@@ -88,70 +99,143 @@ func (n *nightfallTestSuite) TestReviewDiff() {
 	}
 	expectedComments := []*diffreviewer.Comment{&c, &c, &c}
 
+	totalItems := make([]string, numLines*numFiles)
+	for i := 0; i < numLines*numFiles; i++ {
+		totalItems[i] = content
+	}
+	scanRequestItems := make([][]string, numScanReq)
 	for i := 0; i < numScanReq; i++ {
-		mockAPIClient.EXPECT().ScanAPI().Return(mockScanAPI)
-		mockScanAPI.EXPECT().ScanPayload(gomock.Any(), gomock.AssignableToTypeOf(nightfallAPI.ScanRequest{})).Return(scanResp, nil, nil)
+		startIndex := i * maxItemsForAPIReq
+		var endIndex int
+		if len(totalItems) < startIndex+maxItemsForAPIReq {
+			endIndex = len(totalItems)
+		} else {
+			endIndex = startIndex + maxItemsForAPIReq
+		}
+		scanRequestItems[i] = totalItems[startIndex:endIndex]
+	}
+
+	for i := 0; i < numScanReq; i++ {
+		expectedScanReq := client.CreateScanRequest(scanRequestItems[i])
```

Comment:
these loops have the exact same conditions - can we do all this logic inside just one of the loops?